velox/duckdb/conversion/DuckConversion.cpp

```diff
@@ -59,6 +59,11 @@ LogicalType fromVeloxType(const TypePtr& type) {
     auto [precision, scale] = getDecimalPrecisionScale(*type);
     return LogicalType::DECIMAL(precision, scale);
   }
+
+  if (isDateType(type)) {
```

Comment:
Does `type->isDate()` work?
This seems to be the pattern.

Reply:
The isDateType(type) function was modeled as per isIntervalDayTimeType(type) which compares the shared pointer. And there wasn't a isIntervalDayTime function for type so I stuck with this.

The isDecimal(...) variations are doing a dynamic cast, which I think we should avoid. So I stuck with isDateType()

But of course I can add Type.isDate() that does a shared pointer comparison instead.

@mbasmanova : wdyt ?

Reply:
`Date` is a Velox type unlike `intervalDayTime` which is a Presto Type. 
Velox types have the `is##NAME()`, and `as##NAME()` API which I think we should support for Date.

Reply:
intervalDayTime is also a Velox Type, my bad. It should have these API as well. I will fix this.

Reply:
Added type->isDate() function.